ui/src/ScholarReader.tsx

```diff
@@ -38,6 +40,7 @@ class ScholarReader extends React.Component<ScholarReaderProps, State> {
      * are called from outside ScholarReader.
      */
     this.state = {
+      paperId: props.paperId,
```

Comment:
Not that I expect the paper ID to change during an application session, though if it does, would this transfer the paper ID from the props to the state?

Reply:
Ah, good question. 

It depends on whether the component is remounted. I might need to use `getDerivedStateFromProps` to make sure the value is updated.

I'll test this locally and make sure it works properly before merging.

Reply:
It appeared to update w/o a change.  I think that's because updating the URL refreshes the page and re-executes the entry point.

That said I still adopted the method as I think it's good practice, and ensures the value *does* update if we ever do something fancy and change the `paperId` without refreshing the page.